.github/workflows/on-push-to-main.yml

```diff
@@ -0,0 +1,190 @@
+        # ${{ env.DRUPAL_ROOT }} and ${{ env.MODULE_FOLDER }}.
+      - name: Define DRUPAL_ROOT env variable
+        run: echo "DRUPAL_ROOT=$HOME/drupal" >> $GITHUB_ENV
+
+        # Change the module folder according to your needs, it's probably either
+        # modules/contrib/$DRUPAL_MODULE_NAME or modules/custom/$DRUPAL_MODULE_NAME.
+      - name: Set module folder
+        run: |
+          echo "MODULE_FOLDER=$DRUPAL_ROOT/modules/contrib/$MODULE_FOLDER" \
+            >> $GITHUB_ENV
+
+        # Clone Drupal core into $DRUPAL_ROOT folder.
+        # Core version can be set by changing $DRUPAL_CORE_VERSION.
+      - name: Clone drupal
+        run: |
+          git clone --depth 1 --branch "$DRUPAL_CORE_VERSION" \
+            http://git.drupal.org/project/drupal.git/ $DRUPAL_ROOT
+
+        # Override the platform.php config with currently active PHP version.
+        # As of writing this, the composer.json shipped with core sets
+        # platform.php version to 7.3.0, meaning we possibly get dependencies
+        # that won't work with php 8+.
+
+      - name: Override the platform.php version
+        working-directory: ${{ env.DRUPAL_ROOT }}
+        run: composer config platform.php ${{ matrix.php-version }}
+
+        # Set the module folder as a composer repository, so the latest code
+        # is symlinked from $GITHUB_WORKSPACE to modules/ folder.
+      - name: Install the module
+        working-directory: ${{ env.DRUPAL_ROOT }}
+        run: |
+          composer config repositories.0 path $GITHUB_WORKSPACE
+          composer require $MODULE_REPO --ignore-platform-req=php+
+
+        # PHP 8+ requires newer phpunit, use core's composer script
+        # to upgrade it.
+      - name: Upgrade phpunit
+        working-directory: ${{ env.DRUPAL_ROOT }}
+        run: composer run-script drupal-phpunit-upgrade
+
+        # We use drush's build-in webserver to run tests. Make sure
+        # Drush is installed.
+      - name: Install drush
+        working-directory: ${{ env.DRUPAL_ROOT }}
+        run: composer require "drush/drush ^11.0"
+
+      - name: Install PHPCS
+        working-directory: ${{ env.DRUPAL_ROOT }}
+        run: |
+          composer config --no-plugins \
+            allow-plugins.dealerdirect/phpcodesniffer-composer-installer true
+          composer require --dev "drupal/coder"
+
+        # Install Drupal using minimal installation profile and enable the module.
+      - name: Install Drupal
+        working-directory: ${{ env.DRUPAL_ROOT }}
+        run: |
+          php -d sendmail_path=$(which true); vendor/bin/drush --yes -v \
+            site-install minimal --db-url="$SIMPLETEST_DB"
+          vendor/bin/drush en $DRUPAL_MODULE_NAME -y
+
+      - name: Run PHPCS
+        working-directory: ${{ env.DRUPAL_ROOT }}
+        run: |
+          # vendor/bin/phpcs $MODULE_FOLDER --standard=Drupal \
+          #  --extensions=php,module,inc,install,test,info
```

Comment:
Linting. Another thing I'll need to look at when I have more time but is currently not impacting functionality.